### ISSUE TITLE
Found "you've found a bug" error when it's not really

### ISSUE THREAD

**Issue by jwoertink** [crystal-lang/crystal#7530]
This may be related to #7508 or maybe another, but if you just require a file that doesn't exist, crystal tanks. It should still fail, but just with a "No such file or directory" error instead of the "you've found a bug".

https://play.crystal-lang.org/#/r/6gcr

```crystal
require "fail"
```

Crystal 0.27.2 and 0.27.1 throws this message.

```
while requiring "fail" (Exception)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from __crystal_main
  from main
  from __libc_start_main
  from _start
  from ???
Caused by: can't find file 'fail'

If you're trying to require a shard:
- Did you remember to run `shards install`?
- Did you make sure you're running the compiler in the same directory as your shard.yml? (Crystal::CrystalPath::Error)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from __crystal_main
  from main
  from __libc_start_main
  from _start
  from ???
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

Crystal 0.27.0 shows this one:
```
Error in line 1: while requiring "fail": can't find file 'fail'

If you're trying to require a shard:
- Did you remember to run `shards install`?
- Did you make sure you're running the compiler in the same directory as your shard.yml?
```

**Comment by z64** [comment]
Duplicate of #7385 , fixed in #7386

**Comment by jwoertink** [comment]
ah! My search-fu is not strong tonight hahah. I'll close this out. Seems like 0.28.0 will have it fixed. Thanks 😄
